frontend/pages/policies/PolicyPage/PolicyPage.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useState, useEffect, useContext } from "react";
+import { useQuery, useMutation } from "react-query";
+import { InjectedRouter, Params } from "react-router/lib/Router";
+
+// @ts-ignore
+import Fleet from "fleet"; // @ts-ignore
+import { AppContext } from "context/app";
+import { PolicyContext } from "context/policy";
+import { QUERIES_PAGE_STEPS, DEFAULT_POLICY } from "utilities/constants";
+import globalPoliciesAPI from "services/entities/global_policies"; // @ts-ignore
+import hostAPI from "services/entities/hosts"; // @ts-ignore
+import { IPolicyFormData, IPolicy } from "interfaces/policy";
+import { ITarget } from "interfaces/target";
+import { IHost } from "interfaces/host";
+
+import QuerySidePanel from "components/side_panels/QuerySidePanel";
+import QueryEditor from "pages/policies/PolicyPage/screens/QueryEditor";
+import SelectTargets from "pages/policies/PolicyPage/screens/SelectTargets";
+import RunQuery from "pages/policies/PolicyPage/screens/RunQuery";
+import ExternalURLIcon from "../../../../assets/images/icon-external-url-12x12@2x.png";
+
+interface IPolicyPageProps {
+  router: InjectedRouter;
+  params: Params;
+  location: any; // no type in react-router v3
+}
+
+interface IStoredPolicyResponse {
+  policy: IPolicy;
+}
+
+interface IHostResponse {
+  host: IHost;
+}
+
+const baseClass = "policy-page";
+
+const PolicyPage = ({
+  router,
+  params: { id: paramsPolicyId },
```

Comment:
We will need a param for prefilling out a policy name, description, and SQL.

This issues title is: Add the ability to create and edit detailed policies and associated proprietary queries
I think "associated proprietary queries" refers to prefilling it out

Reply:
Oh yes there is a modal for that but, after revisiting, I think I missed that it was suppose to be a part of this PR. Since we're not releasing it yet, @noahtalerman do you mind if we tackle the "Add a policy" modal in another issue for 4.7.0? I can assign myself to it. @RachelElysia I don't think it should stop any other development. Let me know if that's not the case.

Reply:
@martavis This feature has already been pushed to 4.7.0 so no need to rush this PR in for 4.6.0.

Reply:
> do you mind if we tackle the "Add a policy" modal in another issue for 4.7.0

Nope! @martavis please feel free to file a separate issue if you it makes sense to do so.

Reply:
@noahtalerman they are. I am just using this component in both #2938 and #2917 and not as strong with react.router params requiring new pattern implementation. @martavis we can chat post-#2922 merging but just a head's up

Reply:
@martavis -- sorry the modals are my open draft PRs, I was just giving you a heads up more params will be needed to pre-fill out this component